components/remote_settings/src/lib.rs

```diff
@@ -0,0 +1,209 @@
+        ClientConfig {
+            server_url: c.server_url,
+            bucket_name: c.bucket_name,
+            collection_name: c.collection_name,
+        }
+    }
+}
+
+pub struct JsonObject(pub serde_json::Value);
+
+impl UniffiCustomTypeConverter for JsonObject {
+    type Builtin = String;
+
+    fn into_custom(val: Self::Builtin) -> uniffi::Result<Self> {
+        Ok(JsonObject(serde_json::from_str(&val)?))
+    }
+
+    fn from_custom(obj: Self) -> Self::Builtin {
+        serde_json::to_string_pretty(&obj.0).unwrap()
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use mockito::mock;
+
+    #[test]
+    fn test_get_records() {
+        viaduct_reqwest::use_reqwest_backend();
+        let m = mock(
+            "GET",
+            "/v1/buckets/the-bucket/collections/the-collection/records",
+        )
+        .with_body(response_body())
+        .with_status(200)
+        .with_header("content-type", "application/json")
+        .create();
+
+        let config = RemoteSettingsConfig {
+            server_url: Some(mockito::server_url()),
+            bucket_name: Some(String::from("the-bucket")),
+            collection_name: String::from("the-collection"),
+        };
+        let remote_settings = RemoteSettings::new(config).unwrap();
+
+        let resp = remote_settings.get_records().unwrap();
+
+        assert!(are_equal_json(&jpg_attachment(), &resp[0].0));
+        m.expect(1).assert();
+    }
+
+    // This test was designed as a proof-of-concept and requires a locally-run Remote Settings server.
+    // If this were to be included in CI, it would require pulling the RS docker image and scripting
+    // its configuration, as well as dynamically finding the attachment id, which would more closely
+    // mimic a real world usecase.
+    // #[test]
+    #[allow(dead_code)]
+    fn test_download() {
+        viaduct_reqwest::use_reqwest_backend();
+        let config = RemoteSettingsConfig {
+            server_url: Some("http://localhost:8888".to_string()),
+            bucket_name: Some(String::from("the-bucket")),
+            collection_name: String::from("the-collection"),
+        };
+        let remote_settings = RemoteSettings::new(config).unwrap();
+
+        remote_settings
+            .download_attachment_to_path(
+                "d3a5eccc-f0ca-42c3-b0bb-c0d4408c21c9.jpg".to_string(),
+                "test.jpg".to_string(),
+            )
+            .unwrap();
+    }
+
+    fn are_equal_json(s1: &str, s2: &serde_json::Value) -> bool {
+        let s1: serde_json::Value = serde_json::from_str(s1).unwrap();
+        &s1 == s2
+    }
+
+    fn response_body() -> String {
+        format!(
+            r#"
+        {{
+            "data": [
+                {},
+                {},
+                {}
+            ]
+          }}"#,
+            jpg_attachment(),
+            pdf_attachment(),
+            no_attachment()
+        )
+    }
+
+    fn jpg_attachment() -> String {
```

Comment:
does this and the following functions need to be functions? can we define `const` string slices? for example:

```rs
const JSG_ATTACHMENT: &str = r#"
          {
            "title": "jpg-attachment",
            "content": "content",
            "attachment": {
            "filename": "jgp-attachment.jpg",
            "location": "the-bucket/the-collection/d3a5eccc-f0ca-42c3-b0bb-c0d4408c21c9.jpg",
            "hash": "2cbd593f3fd5f1585f92265433a6696a863bc98726f03e7222135ff0d8e83543",
            "mimetype": "image/jpeg",
            "size": 1374325
            },
            "id": "c5dcd1da-7126-4abb-846b-ec85b0d4d0d7",
            "schema": 1677694447771,
            "last_modified": 1677694949407
          }
        "#

```

Reply:
Definitely, I had just been following the convention in my last PR. Thanks for the suggestion!